end-to-end-tests/tst/helpers/resources.py

```diff
@@ -0,0 +1,129 @@
+            return self
+
+    def set_status(self, logical_resource_id, new_status):
+        resource = self.resources.get(logical_resource_id)
+
+        if resource is None:
+            raise ValueError("Resource with logical ID not found - {}".format(logical_resource_id))
+        resource.status = new_status
+        return self
+
+    def remove(self, logical_resource_id):
+        self.resources.remove(logical_resource_id)
+        return self
+
+    def from_list(self, resources_list):
+        resources_from_list = Resources()
+        for stack_resource in resources_list:
+            resources_from_list.add(stack_resource=stack_resource)
+        return resources_from_list
+
+    # helper method for adding tests, call when needed
+    def get_changed_resources(self, expected_resources):
+        diff_resources = dict()
+        for key in self.resources.keys():
+            if key not in expected_resources.resources or not expected_resources.resources.get(
+                key
+            ) == self.resources.get(key):
+                diff_resources[key] = self.resources.get(key)
+            # remove the resource if its same in generated resources and expected resources
+            expected_resources.resources.pop(key, None)
+
+        if expected_resources.resources is not None:
+            diff_resources.update(expected_resources.resources)
+
+        return diff_resources
+
+    def __eq__(self, other):
+        if not isinstance(other, Resources):
+            return False
+        # checking if the keys are same
+        if self.resources.keys() != other.resources.keys():
+            return False
+        for id, resource in self.resources.items():
+            if not (resource == other.resources.get(id)):
+                return False
+        return True
+
+    """
+    Validates and Cleans up LogicalIDs generated by SAM translator.
+    @param logical_id_to_clean
+    @param resource_type
+    @return
+    """
+
+    def clean_logicalid_if_necessary(self, logical_id_to_clean, resource_type):
+        if (
+            ResourceTypes.APIGW_DEPLOYMENT.value == resource_type
+            or ResourceTypes.LAMBDA_VERSION.value == resource_type
+            or ResourceTypes.LAMBDA_LAYER_VERSION.value == resource_type
+        ):
+            if not re.match(self.LOGICALID_WITH_HASH_PATTERN, logical_id_to_clean):
+                raise ValueError(
+                    "{} LogicalID of {} does not follow pattern. It should end with 10 hex characters generated by the SHA".format(
+                        logical_id_to_clean, resource_type
+                    )
+                )
+            return logical_id_to_clean[:-10]  # remove last 10 digits and return
+        return logical_id_to_clean
+
+
+class Resource:
+    # We only care about the following properties when comparing for equality because rest are dynamically
+    # generated by CloudFormation: LogicalResourceId, ResourceStatus, ResourceType
+    logical_id = str()
```

Comment:
Why are these defined globally on Resource?

Reply:
Good catch! We don't need these attributes to be present globally. I have updated the code